src/busctl/busctl.c

```diff
@@ -1202,13 +1206,20 @@ static int message_json(sd_bus_message *m, FILE *f) {
         e[0] = m->header->endian;
         e[1] = 0;
 
+        ts = m->realtime;
+        if (ts == 0)
+                ts = now(CLOCK_REALTIME);
+
+        p = format_timestamp_style(buf, sizeof(buf), ts, TIMESTAMP_US_UTC);
+
         r = json_build(&w, JSON_BUILD_OBJECT(
                 JSON_BUILD_PAIR("type", JSON_BUILD_STRING(bus_message_type_to_string(m->header->type))),
                 JSON_BUILD_PAIR("endian", JSON_BUILD_STRING(e)),
                 JSON_BUILD_PAIR("flags", JSON_BUILD_INTEGER(m->header->flags)),
                 JSON_BUILD_PAIR("version", JSON_BUILD_INTEGER(m->header->version)),
                 JSON_BUILD_PAIR("cookie", JSON_BUILD_INTEGER(BUS_MESSAGE_COOKIE(m))),
                 JSON_BUILD_PAIR_CONDITION(m->reply_cookie != 0, "reply_cookie", JSON_BUILD_INTEGER(m->reply_cookie)),
+                JSON_BUILD_PAIR("timestamp", JSON_BUILD_STRING(p ? p : "unrepresentable")),
```

Comment:
I think `strna(p)` is enough. At least, please drop redundant `p` after `?`, that is, `p ?: "unrepresentable"`.